lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp

```diff
@@ -253,7 +253,7 @@ namespace Js
 
         if (info.GetPropertyString() != nullptr && info.GetPropertyString()->ShouldUseCache())
         {
-            Assert(propertyString == info.GetPropertyString());
+            Assert(info.IsNoCache() || propertyString == info.GetPropertyString());
```

Comment:
I thought it was fine, and would be a nop, but probably better off to add this check as a condition of the if anyway.